core/src/main/scala/kafka/cluster/BrokerEndPoint.scala

```diff
@@ -36,23 +35,6 @@ object BrokerEndPoint {
       case _ => None
     }
   }
-  
-  /**
-   * BrokerEndPoint URI is host:port or [ipv6_host]:port
-   * Note that unlike EndPoint (or listener) this URI has no security information.
-   */
-  def createBrokerEndPoint(brokerId: Int, connectionString: String): BrokerEndPoint = {
-    parseHostPort(connectionString).map { case (host, port) => new BrokerEndPoint(brokerId, host, port) }.getOrElse {
-      throw new KafkaException("Unable to parse " + connectionString + " to a broker endpoint")
-    }
-  }
-
-  def readFrom(buffer: ByteBuffer): BrokerEndPoint = {
```

Comment:
I'm wondering if the `readFrom` is unused, why does `writeTo` still exist? We only need to write, and no need to read? Could you help confirm it? Thanks.

Reply:
Correct.
We can remove additional methods from class.